tests/kafkatest/tests/produce_consume_validate.py

```diff
@@ -50,12 +50,33 @@ def start_producer_and_consumer(self):
         # Start background producer and consumer
         self.consumer.start()
         if (self.consumer_init_timeout_sec > 0):
-            self.logger.debug("Waiting %ds for the consumer to fork.",
+            self.logger.debug("Waiting %ds for the consumer to initialize.",
                               self.consumer_init_timeout_sec)
+            start = int(time.time())
             wait_until(lambda: self.consumer.alive(self.consumer.nodes[0]) is True,
                        timeout_sec=self.consumer_init_timeout_sec,
                        err_msg="Consumer process took more than %d s to fork" %\
                        self.consumer_init_timeout_sec)
+            end = int(time.time())
+            # TODO(apurva): This sleep is a hack. The problem is that the
+            # the JmxTool is spawned before the consumer process is fully
+            # initialized. Hence the connection fails. Retries on connection
+            # failures were added to the JmxTool, but unfortunately, it appears
+            # that the exception does not bubble to the top level, rendering
+            # the retry loop useless.
+            #
+            # We can remove this sleep once we figure out a way to do proper
+            # retries in JMXTool.
+            #
+            # This is covered in KAFKA-4620
```

Comment:
Nit: I think we can make this a bit more specific now that we understand the issue, maybe something like:

```python
# If `JMXConnectorFactory.connect` is invoked during initialisation of the JMX server,
# it may fail to throw the specified IOException back to the calling code. The
# sleep is a workaround that should allow initialisation to complete before we try
# to connect. See KAFKA-4620 for more details.
```

It's a bug in the Java library, so I don't think there's much that we can do in `JmxTool`.